raft/server.cc

```diff
@@ -1888,4 +1892,28 @@ std::ostream& operator<<(std::ostream& os, const server_impl& s) {
     return os;
 }
 
+future<> server_impl::override_snapshot_thresholds() {
+    return utils::get_local_injector().inject(
+        "raft_server_set_snapshot_thresholds", [this](auto& handler) -> future<> {
+            const auto set_parm = [&handler](auto& target, const std::string_view name) -> void {
+                const auto from = handler.get(name);
+                if (from) {
+                    try {
+                        target = boost::lexical_cast<std::remove_reference_t<decltype(target)>>(*from);
+                        logger.info("Applied _config.{}={}", name, *from);
+                    } catch (const boost::bad_lexical_cast& e) {
+                        logger.error("Could not apply a snapshot threshold param: {}, value: {}, error: {}",
+                                     name, *from, e.what());
+                        throw;
```

Comment:
In this case we should crash immediately, because there's a bug in the test. (It passed incompatible type.)
So instead of `logger.error` + `throw`, use `on_internal_error(logger, format("blabla {} bla {}", args...))` -- if we enter this branch, Scylla will crash and the test will fail.